cursor.go

```diff
@@ -521,3 +542,72 @@ func (c *Cursor) extendLocked(response *Response) {
 
 	putResponse(response)
 }
+
+// seekCursor takes care of loading more data if needed and applying pending skips
+func (c *Cursor) seekCursor() error {
+	if c.lastErr != nil {
+		return c.lastErr
+	}
+
+	if len(c.responses) == 0 && c.closed {
+		return errCursorClosed
+	}
+
+	c.applyPendingSkips(false)
```

Comment:
Could you think of a way to reduce the number of calls to `applyPendingSkips`? I feel like it makes the flow slightly more complicated.